lib/settings/view/settings_page.dart

```diff
@@ -557,6 +575,102 @@ class _ExposeOnlineSection extends StatelessWidget with WatchItMixin {
+                if (lastFmEnabled)
+                  Form(
+                    key: formkey,
+                    child: Column(
+                      children: [
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 4),
+                          child: TextFormField(
+                            controller: lastFmApiKeyController,
+                            decoration: InputDecoration(
+                              hintText: l10n.lastfmApiKey,
+                            ),
+                            validator: (value) {
+                              if (value == null || value.isEmpty) {
+                                return l10n.lastfmApiKeyEmpty;
+                              }
+                              return null;
+                            },
+                            onFieldSubmitted: (value) async {
+                              if (formkey.currentState!.validate()) {
+                                di<SettingsModel>().setLastFmApiKey(value);
+                              }
+                            },
+                          ),
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 4),
+                          child: TextFormField(
+                            controller: lastFmSecretController,
+                            decoration: InputDecoration(
+                              hintText: l10n.lastfmSecret,
+                            ),
+                            validator: (value) {
+                              if (value == null || value.isEmpty) {
+                                return l10n.lastfmSecretEmpty;
+                              }
+                              return null;
+                            },
+                            onFieldSubmitted: (value) async {
+                              if (formkey.currentState!.validate()) {
+                                di<SettingsModel>().setLastFmSecret(value);
+                              }
+                            },
+                          ),
+                        ),
+                      ],
+                    ),
+                  ),
+              ],
+            ),
+            trailing: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                CommonSwitch(
+                  value: lastFmEnabled,
+                  onChanged: (v) {
+                    di<SettingsModel>().setEnableLastFmScrobbling(v);
+                  },
+                ),
+                if (lastFmEnabled)
+                  ImportantButton(
+                    onPressed: () async {
+                      if (lastFmApiKeyController.text.isNotEmpty &&
+                          lastFmSecretController.text.isNotEmpty) {
+                        final lastfmua = di<LastFM>() as LastFMUnauthorized;
```

Comment:
I would prefer if you create a LastFMModel here or create a method for this inside the AppModel, which is not perfect but better than to do this all here in this onpressed callback

Reply:
Righto, will move it.